auth/src/main/java/com/firebase/ui/auth/util/signincontainer/SignInDelegate.java

```diff
@@ -232,17 +237,44 @@ private void handleCredential(Credential credential) {
 
     private void startAuthMethodChoice() {
         List<AuthUI.IdpConfig> idpConfigs = mHelper.getFlowParams().providerInfo;
+        Map<String, IdpConfig> providerIdToConfig = new HashMap<>();
+        for (IdpConfig providerConfig : idpConfigs) {
+            providerIdToConfig.put(providerConfig.getProviderId(), providerConfig);
+        }
 
+        List<IdpConfig> visibleProviders = new ArrayList<>();
+        if (mHelper.getFlowParams().isReauth) {
+            // For reauth flow we only want to show the IDPs which the user has associated with
+            // their account.
+            List<String> providerIds = mHelper.getCurrentUser().getProviders();
+            if (providerIds.size() == 0) {
+                // zero providers indicates that it is an email account
```

Comment:
It does seem to be different, asking some of the Firebase folks for more information